src/dotnet/Telemetry.cs

```diff
@@ -78,6 +116,13 @@ public void TrackEvent(string eventName, IDictionary<string, string> properties,
             }
         }
 
+        public void Finish()
+        {
+            if(_trackEventTask != null)
+            {
+                _trackEventTask.Wait(TimeSpan.FromMilliseconds(TimeoutInMilliseconds));
```

Comment:
This could throw if the task gets cancelled. Can you wrap this in a try/catch just in case?